amr-wind/utilities/averaging/TimeAveraging.cpp

```diff
@@ -71,25 +76,39 @@ const std::string& TimeAveraging::add_averaging(
 
     // Create and register new average
     m_averages.emplace_back(
-        FieldTimeAverage::create(avg_type, m_sim, field_name));
+        FieldTimeAverage::create(avg_type, m_sim, m_label, field_name));
     return m_averages.back()->average_field_name();
 }
 
 void TimeAveraging::post_advance_work()
 {
     const auto& time = m_sim.time();
     const auto cur_time = time.new_time();
-
-    // Check if we are within the averaging time period requested by the user
+    const auto cur_step = time.time_index();
+
+    // Check the following:
+    //   1. if we are within the averaging time period requested by the user
+    //   2. if we are phase averaging (i.e., only accumulating the average at a
+    //   certain
+    //      frequency), check to see if we are on a time step in which averaging
+    //      should be performed.  This is useful for simulations with periodic
+    //      behavior, such as regular waves or actuator lines rotating at fixed
+    //      rotor speed.
     const bool do_avg =
-        ((cur_time >= m_start_time) && (cur_time < m_stop_time));
+        (((cur_time >= m_start_time) && (cur_time < m_stop_time)) &&
+         (cur_step % m_frequency == 0));
     if (!do_avg) {
         return;
     }
 
     const amrex::Real elapsed_time = (cur_time - m_start_time);
+    m_last_avg_time = cur_time;
+    amrex::Print() << "TimeAveraging: " << m_label << std::endl;
     for (const auto& avg : m_averages) {
-        (*avg)(time, m_filter, elapsed_time);
+        amrex::Print() << "Accumulating average.  Last accumulation done at: "
+                       << m_last_avg_time << ", elapsed time: " << elapsed_time
+                       << std::endl;
```

Comment:
@mchurchf Is there a benefit to having this print statement within the loop? It seems like it would just be repeating the same information for every avg.